app/main/posts/views/filters/filter-saved-search.directive.js

```diff
@@ -23,22 +23,17 @@ function FilterSavedSearch(SavedSearchEndpoint, _,  $rootScope, ModalService) {
         };
 
         function activate() {
-            ngModel.$render = renderModelValue;
-            if (ngModel.$viewValue.length > 0) {
+            if (ngModel.$viewValue) {
                 scope.selectedSavedSearch = scope.searches[ngModel.$viewValue];
             }
             scope.$watch('selectedSavedSearch', saveValueToView, true);
         }
         function saveValueToView(selectedSavedSearch) {
-            ngModel.$setViewValue(selectedSavedSearch && selectedSavedSearch.id ?
-                selectedSavedSearch.id.toString() : ngModel.$viewValue);
-        }
-
-        function renderModelValue() {
-            scope.selectedSavedSearch = ngModel.$viewValue;
-            // Update savedSearch w/o breaking references used by selectedSavedSearch model
+            if (selectedSavedSearch && selectedSavedSearch.hasOwnProperty('filter')) {
+                PostFilters.setFilters(selectedSavedSearch.filter);
+            }
+            ngModel.$setViewValue(selectedSavedSearch);
```

Comment:
only doing this for the transformers which render the values somewhere else. But I  started thinking, I don't think I actually need it. Don't want to mess too much with things today TBH but I'm not really sure about this.